CI: Switch to macOS-11 as 10.15 will be removed Aug 30th

Geod24 opened the pull request:
We've just been hit by a brownout.

RazvanN7 commented on druntime/posix.mak:
```diff
@@ -58,7 +58,14 @@ endif
+#https://web.archive.org/web/20201213125509/https://www.oreilly.com/library/view/mac-os-x/0596003560/ch05s02.html
+	LINKLIBS=''
+else
+	LINKLIBS=$(if $(findstring $(OS),linux),-L-ldl,) -L-lpthread -lm
```
```suggestion
	LINKLIBS=$(if $(findstring $(OS),linux),-L-ldl,) -L-lpthread -L-lm
```

maxhaton replied:
@Geod24 ?

thewilsonator commented:
We have another brownout, what needs to be done to move this forward?

Geod24 commented:
> We have another brownout, what needs to be done to move this forward?

Was a bit puzzled by the issue, just found out that we need to change Phobos first :upside_down_face: 
https://github.com/dlang/phobos/pull/8527

thewilsonator commented:
Dependant PR merged

thewilsonator commented:
ping

thewilsonator commented:
```
# All hosts are 64 bits but let's be explicit
[3](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:3)
  ./dmd/compiler/src/build.d -j2 MODEL=64
[4](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:4)
  make -C dmd/druntime -f posix.mak -j2 MODEL=64
[5](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:5)
  make -C phobos   -f posix.mak -j2 MODEL=64
[6](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:6)
  # Both version can live side by side (they end up in a different directory)
[7](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:7)
  # However, since clang does not provide a multilib package, only test 32 bits with g++
[8](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:8)
  if [ clang == "g++" ]; then
[9](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:9)
    ./dmd/compiler/src/build.d install -j2 MODEL=32
[10](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:10)
    make -C dmd/druntime -f posix.mak install -j2 MODEL=32
[11](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:11)
    make -C phobos   -f posix.mak install -j2 MODEL=32
[12](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:12)
  fi
[13](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:13)
  shell: /bin/bash -e {0}
[14](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:14)
  env:
[15](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:15)
    DC: dmd
[16](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:16)
    LD_LIBRARY_PATH: /Users/runner/hostedtoolcache/dc/dmd-2.100.0/x64/dmd2/linux/lib64
[17](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:17)
    CC: /Users/runner/work/dmd/dmd/clang+llvm-11.0.0-x86_64-apple-darwin/bin/clang-wrapper
[18](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:18)
    CXX: /Users/runner/work/dmd/dmd/clang+llvm-11.0.0-x86_64-apple-darwin/bin/clang++-wrapper
[19](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:20)
Warning: Syncing file access because of OSX!
[20](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:21)
Warning: Syncing file access because of OSX!
[21](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:22)
ld: library not found for -lpthread
[22](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:23)
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
[23](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:24)
Error: linker exited with status 1
[24](https://github.com/dlang/dmd/runs/7817610235?check_suite_focus=true#step:16:25)
Error: Process completed with exit code 1.
```

ibuclaw commented on .github/workflows/runnable_cxx.yml:
```diff
@@ -236,13 +236,12 @@ jobs:
-        # Note: We need to use this because github.workspace is not stable
-        SCRIPT_FULL_PATH=$(dirname "$0")
-        ${SCRIPT_FULL_PATH}/clang -isystem /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/ $@
+        echo "Wrapper script called with: $@"
+        /usr/bin/cc $@
```
This is the _real_ fix for our custom github action environment
```suggestion
        # Note: We need to use this because github.workspace is not stable
        SCRIPT_FULL_PATH=$(dirname "$0")
        ${SCRIPT_FULL_PATH}/clang -isysroot $(xcrun --show-sdk-path) $@
```

**Or** add `export SDKROOT=$(xcrun --show-sdk-path)` somewhere.

ibuclaw reviewed (changes requested):
The issue isn't the compiler, it's the modified environment that we're running the pipeline in.

See https://github.com/dlang/dmd/pull/14479#issuecomment-1256912541

ibuclaw committed:
GH CI: Set `-isysroot` to the default pipeline SDK path & use OSX11

ibuclaw commented:
@Geod24 all passing now. :-)

Geod24 commented:
@ibuclaw : Thanks ! I rebased on master, trimmed down the commit, and went with a slightly different approach: Instead of setting `SDKROOT`, I just changed the key of the cache to trigger a cache miss (if you look, it was already using `-1` as a suffix for this reason).
Checking if this pass, but if it does, we probably don't need the wrapper scripts (we can just set SDKROOT, which I didn't knew about).

ibuclaw commented:
> @ibuclaw : Thanks ! I rebased on master, trimmed down the commit, and went with a slightly different approach: Instead of setting `SDKROOT`, I just changed the key of the cache to trigger a cache miss (if you look, it was already using `-1` as a suffix for this reason). Checking if this pass, but if it does, we probably don't need the wrapper scripts (we can just set SDKROOT, which I didn't knew about).

Some primer information about SDKROOT/sysroot/syslibroot: https://lists.apple.com/archives/xcode-users/2005/Dec/msg00524.html

I read this comment before settling on SDKROOT: https://trac.macports.org/ticket/57612#comment:18

Geod24 commented:
Reverted back to your diff.

ibuclaw commented on .github/workflows/runnable_cxx.yml:
```diff
@@ -221,7 +221,7 @@ jobs:
       with:
         path: ${{ github.workspace }}/clang+llvm-${{ matrix.cxx-version }}-${{ matrix.arch }}/
-        key: ${{ matrix.cxx-version }}-${{ matrix.arch }}-1
+        key: ${{ matrix.cxx-version }}-${{ matrix.arch }}-3
```
Switch back to -1?

Geod24 replied:
Why? The cache has been propagated, it doesn't really matter.

ibuclaw replied:
I don't know which version of the script this is using. Maybe make it a timestamp just in case? `-20220925`

Geod24 replied:
Changed to use a timestamp, and we can't expire them AFAIK
